docs/workflow/blocks/context_block.rst

```diff
@@ -1,28 +1,44 @@
-Context
-=======
+Context & State
+===============
 
-Use the context block to save data into the global context to make it available for all
+Use the context block to save data into the flow context or global state to make it available for all
 subsequent tasks.
 
+There are two different storage domains that can be written to with this block. 
+
+**Context** is passed from block to block. When processing data in parrallel - during a batch or multiplex flow - context will be different in each branch.
+Blocks will only have access to values written by another block in the same branch. 
+
+**state** is shared by all blocks, and by all flows. When writing to or reading from State, the data is available on two different branches. 
+**state.global** is a datastore that persists for a user's entire session, no matter where they are in the site. 
```

Comment:
Maybe good to say that global is shared by every flow they use, so it's a bad place to put things like sensitive API session keys?

Reply:
We should probably point that out too.

Reply:
Someone might also expect to be able to have state persist across tabs that are both active in memory at the same time. Probably best explaining that won't work?

Reply:
Someone might think local state might be protected. Maybe good to explain it's not currently a security feature and using it as such would not work but permissions are on our roadmap and enforcing access to state may happen later?

Reply:
Additional content added to explain the protection and and persistence.